lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -1084,6 +1084,9 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
 
         String endpoint = hadoopConfig.get(S3_ENDPOINT);
         String region = hadoopConfig.get(S3_REGION);
+        if (region == null) {
+            region = System.getenv("AWS_REGION");
```

Comment:
Personally I would prefer to fall back to the `DefaultAwsRegionProviderChain` https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/regions/providers/DefaultAwsRegionProviderChain.html